docs/acceptance-criteria/uacs-rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,467 @@
+  at the moment when the first member becomes eligible (the moment when the
+  relay entry was requested). +
+  The hard relay entry timeout is a set value (measured in time units).
+
+| Relay request - relay entry timeouts
+| If no entry was successfully submitted until soft relay timeout is reached,
+  all operators in the group should start bleeding and losing their T stake. The
+  amount of lost stake should incrementally increase from 0 (at the moment of
+  soft relay entry timeout) to a fixed value (at the moment of hard relay entry
+  timeout).
+
+| Relay request - relay entry timeouts
+| If no entry was successfully submitted until soft relay timeout is reached,
+  all operators in the group should start bleeding and losing their T stake. The
+  amount of lost stake should incrementally increase from 0 (at the moment of
+  soft relay entry timeout) to a fixed value (at the moment of hard relay entry
+  timeout).
+
+| Relay request - relay entry timeouts
+| Anyone can send a transaction reporting hard relay entry timeout.
+
+| Relay request - relay entry timeouts
+| Upon noticing the hard relay entry timeout notification, the random beacon
+  contract should determine its validity.
+
+| Relay request - relay entry timeouts
+| When determining validity of the hard relay entry timeout notification, the
+  following aspects should be taken into account: +
+  - does it reference the existing relay entry request? (if not, then
+  notification invalid) +
+  - has the hard relay entry timeout passed? (if not, then notification invalid) +
+
+| Relay request - relay entry timeouts
+| If the hard relay entry timeout notification was deemed justified, then: +
+  - the signing group should get terminated +
+  - the process of choosing a signing group and tasking it with a relay entry
+  generation should be repeated.
+
+| Relay entry utilisation
+| Applications wanting to use a relay entry should submit another transaction
+  using the relay entry value previously set by the random beacon.
+
+| Relay entry utilisation
+| Smart contract consuming new relay entry needs to implement
+  `IRandomBeaconConsumer` interface.
+
+| Group creation
+| New groups should be crated with a fixed frequency of relay request (every
+  `N`-th relay entry request should result in the creation of a new group).
+
+| Group creation
+| If according to group creation frequency the relay entry request is the one
+  that should trigger new group creation, the creation should be triggered once
+  a new relay entry appears on the chain (all off-chain clients should start the
+  process of creation of new group using the new entry value and a view
+  sortition pool function call).
+
+| Group creation
+| Frequency must be rare enough to leave the time for the group creation and
+  then (when the pool gets unlocked) for joining operators to pools.
+
+| Group creation
+| Group creation start transaction should be embedded into relay entry request
```

Comment:
The group creation should be embedded into the relay entry submission transaction. Looks like there is a mistake in the RFC.

Reply:
@lukasz-zimnoch, so the new groups will be created with some specific frequency of the relay entry *requests*, but the trigger for the creation of group will be included in the relay entry *submission* tx, right?

So let's say the new groups are created every 3 relay requests.
req_x - group G1 submitted the relay entry, no creation of a new group requested
req_x+1 - group G2 submitted the relay entry, no creation of a new group requested
req_x+2 - group G3 submitted the relay entry, requested the creation of a new group
req_x+3 - group G4 submitted the relay entry, no creation of a new group requested
req_x+4 - group G5 submitted the relay entry, no creation of a new group requested
req_x+5 - group G6 failed to submit the relay entry and was terminated, group G7 was selected to create the entry, G7 submitted the relay entry and requested the creation of a new group
(G1-G7 do not need to be different groups)

Do I get that right?

Reply:
Yeah. Each request has a `requestId` starting from `1`. If `requestId mod groupCreationFrequency == 0`, the submission of relay entry corresponding to that request will trigger group creation by emitting the `DkgStarted` event.